apps/web-staking/src/app/hooks/useGetRedemptions.ts

```diff
@@ -4,61 +4,79 @@ import { useAccount } from 'wagmi';
-            const refreshedRedemptions = await refreshRedemptions(getNetwork(chainId), address, redemptions);
-            setRedemptions(refreshedRedemptions);
-        } catch (error) {
-            setRedemptionsLoading(false);
-            console.error('Failed to update user redemptions:', error);
-        }
-    }, [address, chainId, redemptions, setRedemptions]);
-
-    const loadAllUserRedemptions = useCallback(async () => {
-        try {
-            if (!address || !chainId) return;
-            const redemptionsFromChain = await getAllRedemptions(getNetwork(chainId), address);
-            setRedemptions(redemptionsFromChain);
-            setRedemptionsLoaded(true);
-        } catch (error) {
-            setRedemptionsLoaded(false);
-            setRedemptionsLoading(false);
-            console.error('Failed to load user redemptions:', error);
-        }
-    }, [address, chainId, setRedemptions, setRedemptionsLoaded]);
 
     const loadRedemptions = useCallback(async () => {
-        const now = Date.now();
-        if (now - lastLoadedRef.current < 1000) {
+        if (!address || !chainId || !isConnected) {
+            console.log("Cannot load redemptions - missing requirements", { address, chainId, isConnected });
             return;
         }
-        setRedemptionsLoading(true);
-        lastLoadedRef.current = now;
-        if (redemptionsLoaded) {
-            await updateUserRedemptions();
-        } else {
-            await loadAllUserRedemptions();
+        const now = Date.now();
+        // Do not reload if already loaded in the last second
+        if (now - lastLoadedRef.current > 1000) {
+            lastLoadedRef.current = now;
+
+            setRedemptionsLoading(true);
+            try {
+                // If address changed or not loaded yet, do a fresh load
+                if (loadedAddress !== address || !redemptionsLoaded) {
+                    const redemptionsFromChain = await getAllRedemptions(getNetwork(chainId), address);
+                    setRedemptions(redemptionsFromChain);
+                    setRedemptionsLoaded(true);
+                } else {
+                    const refreshedRedemptions = await refreshRedemptions(getNetwork(chainId), address, redemptions);
+                    setRedemptions(refreshedRedemptions);
+                }
+            } catch (error) {
+                console.error('Failed to load/update redemptions:', error);
+                setRedemptionsLoaded(false);
+                setRedemptions(INITIAL_REDEMPTIONS);
+            } finally {
+                setRedemptionsLoading(false);
+            }
         }
-        setRedemptionsLoading(false);
-    }, [redemptionsLoaded, updateUserRedemptions, loadAllUserRedemptions]);
+    }, [
+        address,
+        chainId,
+        isConnected,
+        loadedAddress,
+        redemptionsLoaded,
+        redemptions,
+        setRedemptions,
+        setRedemptionsLoaded,
+    ]);
```

Comment:
Im not sure useCallback() actually makes sense here with this list of dependencies. Seems we will have recreate the function with any state of the component anyway ?

Do I not see something here or should we change this to a regular function with needed state as input args ?

Reply:
You are correct, useCallback doesn't add any optimizations here. It has been removed in favor of a function that accepts arguments.